apps/dashboard/src/components/ai-suggestions/ai-suggestions-dialog.tsx

```diff
@@ -0,0 +1,170 @@
+import { AiWorkflowContext, GenerateContentResponse } from '@/api/ai';
+import { Dialog, DialogContent, DialogDescription, DialogHeader, DialogTitle } from '@/components/primitives/dialog';
+import { useGenerateAiContent } from '@/hooks/use-generate-ai-content';
+import { StepTypeEnum } from '@novu/shared';
+import { useCallback, useMemo, useState } from 'react';
+import { RiCloseLine, RiCornerDownLeftLine } from 'react-icons/ri';
+import { PROMPT_SUGGESTIONS_BY_STEP } from './constants';
+import { ContentPreview } from './content-preview';
+import { ErrorState } from './error-state';
+import { InitialState } from './initial-state';
+import { LoadingState } from './loading-state';
+import { safeParseJson } from './utils';
+
+type AiSuggestionsDialogProps = {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  stepType: StepTypeEnum;
+  context?: AiWorkflowContext;
+  previewPayload?: string;
+  hasExistingContent?: boolean;
+  onInsert: (content: GenerateContentResponse['content'], suggestedPayload?: Record<string, string>) => void;
+};
+
+export function AiSuggestionsDialog({
+  open,
+  onOpenChange,
+  stepType,
+  context,
+  previewPayload,
+  hasExistingContent = false,
+  onInsert,
+}: AiSuggestionsDialogProps) {
+  const [prompt, setPrompt] = useState('');
+  const [visibleSuggestionsCount, setVisibleSuggestionsCount] = useState(2);
+  const { isLoading, isError, error, lastResponse, generate, reset } = useGenerateAiContent({
+    stepType,
+    context,
+  });
+  const hasResponse = !!lastResponse;
+  const parsedPayload = useMemo(() => (previewPayload ? safeParseJson(previewPayload) : undefined), [previewPayload]);
+
+  const handleClose = useCallback(() => {
+    onOpenChange(false);
+    setPrompt('');
+    setVisibleSuggestionsCount(2);
+    reset();
+  }, [onOpenChange, reset]);
+
+  const handleSubmit = useCallback(() => {
+    if (prompt.trim()) {
+      generate(prompt);
+    }
+  }, [prompt, generate]);
+
+  const handleSuggestionClick = useCallback(
+    (suggestion: string) => {
+      setPrompt(suggestion);
+      generate(suggestion);
+    },
+    [generate]
+  );
+
+  const handleFeelingLucky = useCallback(() => {
+    const suggestions = PROMPT_SUGGESTIONS_BY_STEP[stepType] || [];
+    const randomSuggestion = suggestions[Math.floor(Math.random() * suggestions.length)];
+    setPrompt(randomSuggestion);
+    generate(randomSuggestion);
+  }, [generate, stepType]);
+
+  const handleLoadMore = useCallback(() => {
+    const suggestions = PROMPT_SUGGESTIONS_BY_STEP[stepType] || [];
+    setVisibleSuggestionsCount((prev) => Math.min(prev + 2, suggestions.length));
+  }, [stepType]);
+
+  const handleInsert = useCallback(() => {
+    if (lastResponse?.content) {
+      onInsert(lastResponse.content, lastResponse.suggestedPayload);
+      handleClose();
+    }
+  }, [lastResponse, onInsert, handleClose]);
```

Comment:
The insert action should force storing the workflow, please take a look at:
```
const { saveForm } = useSaveForm();
```

Reply:
Done 👍🏼